pm/recipient.go

```
@@ -159,51 +163,76 @@ func (r *recipient) ReceiveTicket(ticket *Ticket, sig []byte, seed *big.Int) (st
 		}
 	}
 
-	return sessionID, won, r.acceptTicket(ticket, sig, recipientRand)
+	if err := r.updateSenderNonce(recipientRand, ticket.SenderNonce); err != nil {
+		return sessionID, won, err
+	}
+
+	// check advertised params aren't expired
+	latestBlock := r.tm.LastSeenBlock()
+	if ticket.ParamsExpirationBlock.Cmp(latestBlock) < 0 {
+		return sessionID, won, ErrTicketParamsExpired
+	}
+
+	if !r.validRand(recipientRand) {
```

Previously, this check occurred before the `updateSenderNonce()` call which allows the method to fail early and not bother doing state updates if the recipientRand was already revealed. Can we do that here as well?

---

fixed in https://github.com/livepeer/go-livepeer/pull/1383/commits/3e727d0c0ee87fff1c658785375198cb5b40c3e9